coix/algo.py

```diff
@@ -215,7 +215,7 @@ def body_fun(i, q):
       p, q = extend(targets(i + 1), reverses(i)), compose(forwards(i), q)
       return propose(p, q, loss_fn=avo_loss, detach=True)
 
-    return fori_loop(0, num_targets - 1, body_fun, targets(0))
+    return fori_loop(0, num_targets - 1, body_fun, body_fun(0, targets(0)))
```

Comment:
You are correct, it should be `fori_loop(1, ...)`. I made the change a while ago because I had an issue with the structure of the initial loop argument when running AVO. Looking at it again it seems that the change is indeed not needed. The `propose` returned by the loop body returns the return value of `targets(i + 1)` when executed, which should have the same structure as the return value of `targets(0)` (in most use cases).

Reply:
Oh I see. I had another local change that made `extend` return the concatenated return value of both programs, which is probably why I thought the change is necessary. So the change is definitely not needed!